src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,26 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleDeleteConfirmSubmit) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(
+      "#confirm-image-delete-popup-form"
+    );
+    this._handleDeleteConfirmSubmit = handleDeleteConfirmSubmit;
+  }
+
+  // this function accepts an function as an argument, and assigns it to this
+  setSubmitAction(action) {
+    this._handleSubmitCallback = action;
+  }
+
```

Comment:
Needs correction: add the method for rendering loading and use it in index.js

Reply:
I'm not sure what you mean here. There is no rendering occurring related to this popup; it only removes a card from the DOM and the server. Or are your referring to changing the button text while it is running?

Reply:
Oh... I see, I didn't even think of that being rendering.

Reply:
Yes, I'm about changing the button text. It's needed to add a method for that